its/ruling/src/test/resources/eclipse-jetty/java-S3824.json

```diff
@@ -1,4 +1,7 @@
 {
+'org.eclipse.jetty:jetty-project:jetty-jmx/src/main/java/org/eclipse/jetty/jmx/ConnectorServer.java':[
+127,
```

Comment:
I'm a bit surprised that we are not catching the exact same pattern a few lines above (lines 120 and 125). I feel it's due to the fact that we are using constants as `MemberSelectExpressionTree`, which should be exactly the same.

Can you check that if the problem is on the SE engine side (the way we associate constants with SVs) or if it comes from the rule itself?

If it comes from the SE engine, then we need a ticket.

Reply:
I checked and we do not report this issues because SV associated with key is different in containsKey and in put, despite it's actually the same constant. Also true for Enums and reproduces with Map.get()

It seems to come from engine :( Will create a ticket.

Reply:
Created a ticket https://jira.sonarsource.com/browse/SONARJAVA-3626